hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/ActionRepository.java

```diff
@@ -101,20 +100,33 @@
     List<Action> findByTargetAndActiveOrderByIdAsc(JpaTarget target, boolean active);
 
     /**
-     * Retrieves the oldest {@link Action} that is active and referring to the
-     * given {@link Target}.
-     *
-     * @param sort
-     *            order
+     * Retrieves the active {@link Action}s with the highest weights that refer
+     * to the given {@link Target}. If {@link Action}s have the same weight they
+     * are ordered ascending by ID (oldest ones first).
```

Comment:
The "oldest ones first" does not make sense to me because the actions cannot have the same ID?

Reply:
removed it